pkg/agent/kubernetes.go

```diff
@@ -94,17 +97,166 @@ func patchNodeStatus(cli *k8sclient.Clientset, fields map[string]string) error {
 	return err
 }
 
-// watchConfigMap watches changes in a ConfigMap object
-func watchConfigMap(cli *k8sclient.Clientset, ns string, name string) (watch.Interface, error) {
-	listOpts := meta_v1.ListOptions{
-		FieldSelector: "metadata.name=" + name,
+// watch is a wrapper around the k8s watch.Interface
+type watch struct {
+	parent  *watcher
+	kind    string
+	ns      namespace
+	name    string
+	openfn  func(namespace, string) (k8swatch.Interface, error)
+	queryfn func(namespace, string) (interface{}, error)
+	stop    chan struct{}
+	events  chan k8swatch.Event
+}
+
+type openFn func(ns namespace, name string) (k8swatch.Interface, error)
+type queryFn func(ns namespace, name string) (interface{}, error)
+
+const (
+	// SyntheticMissing is a synthetic initial event for currently non-existent object.
+	SyntheticMissing = k8swatch.EventType("SyntheticMissing")
+)
+
+func newWatch(parent *watcher, kind string, ns namespace, open openFn, query queryFn) *watch {
+	return &watch{
+		parent:  parent,
+		kind:    kind,
+		ns:      ns,
+		stop:    make(chan struct{}),
+		events:  make(chan k8swatch.Event),
+		openfn:  open,
+		queryfn: query,
 	}
-	watch, err := cli.CoreV1().ConfigMaps(ns).Watch(listOpts)
-	if err != nil {
-		return nil, agentError("failed to create a watch for configmaps in ns %q: %v", ns, err)
+}
+
+// configNodeWatch creates a watch for k8s Node
```

Comment:
`newNodeWatch` instead of `configNodeWatch`

Reply:
Thanks, fixed.